saleor/plugins/avatax/__init__.py

```diff
@@ -383,8 +382,15 @@ def get_order_lines_data(
             continue
         product = line.variant.product
         product_type = line.variant.product.product_type
-        tax_code = retrieve_tax_code_from_meta(product, default=None)
-        tax_code = tax_code or retrieve_tax_code_from_meta(product_type)
+
+        tax_code = None
+        if product.tax_class:
+            tax_code = retrieve_tax_code_from_meta(product.tax_class, default=None)
+        elif product_type.tax_class:
+            tax_code = retrieve_tax_code_from_meta(product_type.tax_class)
+        else:
+            tax_code = DEFAULT_TAX_CODE
```

Comment:
It seems like this part could be a separate function as it's also in `generate_request_data_from_checkout_lines` method.